systems/plants/@RigidBodyManipulator/forwardKin.m

```diff
@@ -1,4 +1,4 @@
-function [x,J,dJ] = forwardKin(obj,kinsol,body_or_frame_ind,pts,rotation_type, base_or_frame_ind)
+function [x,J,dJ] = forwardKin(obj,kinsol,body_or_frame_ind,pts,rotation_type, base_or_frame_ind, options)
```

Comment:
My suggestion would be to change to the new format, with options coming right after points.  To transition, we should check if the options is numeric (instead of a struct); if it is then we warnOnce and then treat it as if it was rotation_type.

In my view the mandatory arguments are kinsol, body_or_frame_id, and pts...  (it wouldn't make sense to without any of those) everything else as an option?




On Feb 7, 2015, at 12:21 PM, Twan Koolen <notifications@github.com> wrote:

> In systems/plants/@RigidBodyManipulator/forwardKin.m:
> 
> > @@ -1,4 +1,4 @@
> > -function [x,J,dJ] = forwardKin(obj,kinsol,body_or_frame_ind,pts,rotation_type, base_or_frame_ind)
> > +function [x,J,dJ] = forwardKin(obj,kinsol,body_or_frame_ind,pts,rotation_type, base_or_frame_ind, options)
> rotation_type is in the options structure, but I didn't want to change existing functionality yet. If you think we should I can change it.
> Should I also change body_or_frame_ind to body_or_frame_id then?
> My thoughts were to have base_or_frame_ind not be an option but an actual argument just like body_or_frame_ind. I'm not advocating this, but why not move body_or_frame_ind to the options struct too then?
> 
> —
> Reply to this email directly or view it on GitHub.
>